README.md

```diff
@@ -62,6 +61,107 @@ In a separate directory invoke the generator via
 yo ibm-service-enablement
 ```
 
+## Adding new services
+
+To add a new services you will need to follow these steps.
+
+Create a new composed generated by adding a service folder under generators with the prefix service- then the service id of your new service. To illustrate, if service called S3 needs to be added the folder would be called service-s3.
+
+Under this service there is one folder named templates where you will store your templates per language. Currently, we support java-liberty, java-spring, node, python, and swift. If there is a language that you do not support make an issue.
+
+Add an index.js file using the following structure
+
+```
+'use strict';
+const BaseGenerator = require('../lib/generatorbase');
+const SCAFFOLDER_PROJECT_PROPERTY_NAME = ''; //the key name for your service provided to you by our scaffolder team
+const CLOUD_FOUNDRY_SERVICE_NAME = ''; // the key name for your service provided to you by Cloud Foundry
+const CUSTOM_SERVICE_KEY = '' // a custom key name for your service that will be used if you prefer not to use the SCAFFOLDER_PROJECT_PROPERTY_NAME
+const CUSTOM_CRED_KEYS = []; // an array of custom credentila key names for a service
+
+const config = {
+	cloudFoundryIsArray: true,
+	mappingVersion: 1
+};
+
+module.exports = class extends BaseGenerator {
+	constructor(args, opts) {
+		super(args, opts, SCAFFOLDER_PROJECT_PROPERTY_NAME, CLOUD_FOUNDRY_SERVICE_NAME, CUSTOM_SERVICE_KEY, CUSTOM_CRED_KEYS);
+	}
+
+	initializing() {
+		return super.initializing();
+	}
+
+	configuring() {
+		return super.configuring(config);
+	}
+
+	writing() {
+		return super.writing();
+	}
+};
+```
+
+* The **SCAFFOLDER_PROJECT_PROPERTY_NAME** is provided by the IBM Bluemix User Experience. If your service does not exist please make an issue
+so we can accomodate your service. If your service is provisioned using Cloud Foundry set the *CLOUD_FOUNDRY_SERVICE_NAME* using the service name
+from **VCAP_SERVICES** (e.g `cloudantNoSQLDB`).
+
+* The **CLOUD_FOUNDRY_SERVICE_NAME** is provided by **CLOUD FOUNDRY** via the **VCAP_SERVICES** when you 
+provision a service. 
+
+* The **CUSTOM_SERVICE_KEY** is the service key that can be used in place of the keys provided by Scaffolder. An example of these keys can be 
+found under `test/resources/mappings.json` Note 
+
+* The **CUSTOM_CRED_KEYS** is an array of credential keys for each service. If this is left empty the keys then the service will use the credentials keys given by Scaffolder. 
```

Comment:
I think there are typos here: "If this is left empty the keys then the service will use the credentials keys given by Scaffolder. "

"If this is left empty, the service will use the credential keys given by Scaffolder. "  <-- is that what you mean?

Is Scaffolder always capitalized?